src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -680,10 +683,14 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded, Immu
             /// known to be correct for the given state. This would be non-null in cases where we had computed everything and
             /// ran generators, but then the compilation was garbage collected and are re-creating a compilation but we
             /// still had the prior generated result available.</param>
+            /// <param name="nonAuthoritativeGeneratedDocuments">The generated documents from a previous pass which may
+            /// or may not be correct for the current compilation. These states may be used to access cached results, if
+            /// and when applicable for the current compilation.</param>
             private async Task<CompilationInfo> FinalizeCompilationAsync(
                 SolutionState solution,
                 Compilation compilation,
                 ImmutableArray<SourceGeneratedDocumentState>? authoritativeGeneratedDocuments,
+                ImmutableArray<SourceGeneratedDocumentState> nonAuthoritativeGeneratedDocuments,
```

Comment:
Need @jasonmalinowski 's views on this.  I forget what FinalizeCompilation means.  I *thought* it had meant: move the compilation to the final state.  Which means it seems odd to be using incorrect states here.  Is that not the case?  Is FinalizeCompilation for a different purpose?

Reply:
still need @jasonmalinowski 's input here.

Reply:
So yes, FinalizeCompilation is the final (hah!) called when we're producing a compilation. I think what might address your concern is if instead of `nonAuthoritativeGeneratedDocuments` it was called `priorGeneratedDocumentsThatMayBeReused` or something along those lines to clarify that:

1. These are from a prior run, and _may_ be stale/old.
2. It's not required that they are actually consumed.